### PR TITLE
Journalist can sign in

### PR DESCRIPTION
**Pull request by erikbjoern**
# User story 
[PT link](https://www.pivotaltracker.com/story/show/172888072)
```
As a journalist
In order to create an article,
I would like to be able to login as a staff member
```

# Feature
* Adds login to staff client
* Requires staff to log in, in order to access internal pages 
* - Sends credentials in headers when creating article
* - Redirects from/to login page when logging in/out

# Additions in this PR:
* `LoginForm` component, visible on landing page
* Semantic `Menu` component in `Header` component
* `Auth` module from JtockAuth
* Additional styling

## Packages added:
* react-router-dom
* j-tock-auth

# Screenshots
Screenshot 2020-05-25 at 20 22 35
Screenshot 2020-05-25 at 20 28 41

### CODE REVIEW

File: src/components/LoginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    try {
+      const response = await auth.signIn(e.target.email.value, e.target.password.value)
+      props.setUid(response.data.uid)
+      props.setAuthenticated(true)
+    } catch(error) {
+      setMessage(error.response.data.errors[0])
+    }
+  }
+
+  const redirect = props.authenticated && <Redirect to={{pathname: '/write'}}/>
```
Comment (kermit-klein) [inline src/components/LoginForm.jsx]: Nice one!!!

**Review by kermit-klein — APPROVED** [review]

File: src/components/Header.jsx [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
+  const logOut = async () => {
+    try {
+      await auth.signOut();
+      props.setAuthenticated(false);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const redirect = !props.authenticated && <Redirect to={{ pathname: "/" }} />;
```
Comment (ViaMarcus) [inline src/components/Header.jsx:16]: Brilliant putting this in header, much simpler than the ProtectedRoute

**Review by ViaMarcus — APPROVED** [review]
Nice work

**Review by PaulineBA — APPROVED** [review]
Nice!

File: src/App.jsx [hunk 3]
```diff
@@ -1,13 +1,56 @@
...
+      <Header
+        uid={uid}
+        authenticated={authenticated}
+        setAuthenticated={setAuthenticated}
+      />
+      <Switch>
+        <Route
+          exact
+          path="/"
+          render={() => (
```
Comment (designerofthing) [inline src/App.jsx:40]: I don't understand this render in the routes

File: src/components/Header.jsx (hunk 2 above)
Comment (erikbjoern) [reply src/components/Header.jsx:16]: Yes! I put it there just to redirect on logout, then I realised how genius it was :D

**Review by designerofthing — APPROVED** [review]

**Review by jysmys — APPROVED** [review]

**Pull request merged**